src/core/ready.js

```diff
@@ -1,17 +1,16 @@
 define( [
 	"../core",
-	"../var/document",
-	"../deferred"
+	"../var/document"
 ], function( jQuery, document ) {
 
-// The deferred used on DOM ready
-var readyList;
+var readyCallbacks = [],
+	readyFiring = false,
+	whenReady = function( fn ) {
+		readyCallbacks.push( fn );
```

Comment:
As long as you duplicate it in the redefinition of `whenReady`. Doing it this way is just conceptually simpler, and it's not like the function test is expensive.